File: BOJ/단계별 문제/7. 문자열/10. 그룹 단어 체커/dp.py
T = int(input())
ans = 0

for _ in range(T):
    word = input()
    appeared = [0]*(ord('z')-ord('a')+1)
    valid = 1

    p = word[0]
    appeared[ord(p)-ord('a')] = 1

    for x in word[1:]:
        if not appeared[ord(x)-ord('a')]:
            appeared[ord(x)-ord('a')] = 1

        elif x != p:
            valid = 0
            break

        p = x

    if valid:
        ans += 1

print(ans)

File: BOJ/단계별 문제/7. 문자열/10. 그룹 단어 체커/dp.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
    int T, ans = 0;
    string word;
    
    cin >> T;

    while (T--) {
        char p;
        bool appeared['z'-'a'+1] = {0};
        bool valid = 1;

        cin >> word;
        p = word[0];
        appeared[p-'a'] = 1;

        for (int i=1; i<word.length(); i++) {
            if (!appeared[word[i]-'a'])
                appeared[word[i]-'a'] = 1;

            else if (word[i] != p) {
                valid = 0;
                break;
            }

            p = word[i];
        }

        if (valid)
            ans++;
    }

    cout << ans;
    return 0;
}